network/network.go

```diff
@@ -307,6 +307,7 @@ func (c *Collector) calculateNeighbours(n int, ip string, addresses []string) []
 				k = k % len(addresses)
 				neighbours = append(neighbours, addresses[k])
 			}
+			break
```

Comment:
There is only 1 `ip` in `addresses`, once it is found doesn't make sense to keep searching for loop. 

Probably next to no impact but might save a couple small cycles.

Reply:
I don't _think_ we can assume there will only be one. There should be multiple service endpoints